src/lib.rs

```diff
@@ -700,10 +715,26 @@ macro_rules! impl_try_from {
 	    )*
     }
 }
-impl_try_from! {
+impl_i24_try_from_primitive! {
 	u32 i32 usize isize u64 i64 u128 i128
 }
 
+macro_rules! impl_primitive_try_from_i24 {
+    ($($target_type:ty)*) => {
+	    $(
+		    impl core::convert::TryFrom<i24> for $target_type {
+			    type Error = ParseI24Error;
+			    fn try_from(value: i24) -> Result<Self, Self::Error> {
+				    Self::try_from(value.to_i32()).map_err(|_| ParseI24Error::OutOfRange)
+			    }
```

Comment:
`ParseI24Error` will be rolled out completely with #8, and again with the bad code gen